### ISSUE TITLE
No typesafety with CRUD operations

### ISSUE THREAD

**Issue by johannb75** [RobinBuschmann/sequelize-typescript#904]
Hi, 
Thanks for this awesome library.
I want to use `sequelize-typescript` but when I started I faced an issue:
```ts
import { Column, Model, Table } from "sequelize-typescript";

@Table
export class User extends Model {
	@Column
	nickname: string;
}

User.findAll({
	attributes: ["blabla"],
});
```

This doesn't throw an error even tho `blabla` was never defined.
I have the exact same problem when creating a user:
```ts
User.create({
	blabla: "",
        nickname: 88
});
``` 
It is totally accepted by the library even tho `blabla` was never defined and  `nickname` is a `number` and not  a string like defined in the model.
That is the first problem but because of that, the IDE doesn't provide any **autocompletion**.

**Those 2 conditions are the main things Typescript was made for.**

**Comment by johannb75** [comment]
Just figured out that I could do:
```ts
@Table
export class User extends Model<User> {
     @Column
     nickname: string
}
```
Please add this to the docs to make it clear to anyone.

**Comment by johannb75** [comment]
After some testing,  I found out that this option was terrible.  Can you please provide a clean and documented way of being type safe with those operations.

**Comment by johannb75** [comment]
After some testing,  I found out that this option was terrible.  Can you please provide a clean and documented way of being type safe with those operations.

**Comment by RoelVB** [comment]
I would recommend using the [V6 model definitions](https://github.com/RobinBuschmann/sequelize-typescript#v6-model-definition-more-strict), but this isn't implemented in v2.0.0 yet. But PR #900 should fix this.

I'm using the V6 model definitions already, it does work, but I had to ignore some TS errors.

**Comment by johannb75** [comment]
Hi @RoelVB,
Could you give an example of what you mean by `V6 model definitions`.

**Comment by RoelVB** [comment]
@johannb75 It's in the [readme](https://github.com/RobinBuschmann/sequelize-typescript#more-strict)

**Comment by johannb75** [comment]
Well it's pretty bad... I get a lot of errors
